e2e/plugin/fixtures/fixtures.go

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package fixtures
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/pkg/apis/greenhouse/v1alpha1"
+	"github.com/cloudoperators/greenhouse/pkg/test"
+)
+
+func PreparePluginDefinition(name, namespace string, opts ...func(*greenhousev1alpha1.PluginDefinition)) *greenhousev1alpha1.PluginDefinition {
```

Comment:
@IvoGoman - Why don't we declaratively maintain this in `testdata` like for organization? instead of doing it in code? It also mimics `kubectl apply`... it is possible to convert any greenhouse yaml into it's specific types by using `utils.FromYamlToK8sObject`

An advantage would be to in future to use tools like [kuttl](https://kuttl.dev/docs/#pre-requisites) to run smoke tests without having to write code.

Reply:
Done

Reply:
@gciezkowski-acc thanks but let’s wait for a response from @IvoGoman before doing this.

Reply:
The main motivation was to have one place to create these test objects, to not have much boilerplate and to only need to make updates at one place. IMO, this makes reading/debugging the tests easier, as everything is available in code.
Since we already have extensive tests written with Ginkgo/Gomega as a test framework and they work nicely with this setup, I would like to avoid using multiple test frameworks within the code base.

In case we decide in the future that we need YAMLs to execute tests, we could also generate the files by using `utils.FromK8sObjectToYaml(..)`